Calibration/HcalAlCaRecoProducers/plugins/AlCaIsolatedBunchSelector.cc

```diff
@@ -0,0 +1,182 @@
+  static  void globalEndJob(const AlCaIsolatedBunch::Counters* counters);
+  static  void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  
+private:
+  virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
+  virtual void endRun(edm::Run const&, edm::EventSetup const&) override;
+  
+  // ----------member data ---------------------------
+  HLTConfigProvider             hltConfig_;
+  std::string                   trigName_;
+  edm::InputTag                 triggerEvent_, theTriggerResultsLabel_;
+  std::string                   processName_;
+  unsigned int                  nRun_, nAll_, nGood_;
+  edm::EDGetTokenT<trigger::TriggerEvent>  tok_trigEvt_;
+  edm::EDGetTokenT<edm::TriggerResults>    tok_trigRes_;
+};
+
+//
+// constructors and destructor
+//
+AlCaIsolatedBunchSelector::AlCaIsolatedBunchSelector(const edm::ParameterSet& iConfig, const AlCaIsolatedBunch::Counters* count) :
+  nRun_(0), nAll_(0), nGood_(0) {
+  //now do what ever initialization is needed
+  trigName_               = iConfig.getParameter<std::string>("TriggerName");
+  processName_            = iConfig.getParameter<std::string>("ProcessName");
+  triggerEvent_           = iConfig.getParameter<edm::InputTag>("TriggerEventLabel");
+  theTriggerResultsLabel_ = iConfig.getParameter<edm::InputTag>("TriggerResultLabel");
+
+  // define tokens for access
+  tok_trigEvt_  = consumes<trigger::TriggerEvent>(triggerEvent_);
+  tok_trigRes_  = consumes<edm::TriggerResults>(theTriggerResultsLabel_);
+
+  edm::LogInfo("AlCaIsoBunch") << "Input tag for trigger " << triggerEvent_
+			       << " and results " << theTriggerResultsLabel_
+			       << " with trigger name " << trigName_ 
+			       << " and process " << processName_ << std::endl;
+}
+
+AlCaIsolatedBunchSelector::~AlCaIsolatedBunchSelector() {}
+
+//
+// member functions
+//
+
+// ------------ method called on each new Event  ------------
+bool AlCaIsolatedBunchSelector::filter(edm::Event& iEvent, 
+				     edm::EventSetup const& iSetup) {
+  bool accept(false);
+  ++nAll_;
+#ifdef DebugLog
+  edm::LogInfo("AlCaIsoBunch") << "Run " << iEvent.id().run() << " Event " 
+			       << iEvent.id().event() << " Luminosity " 
+			       << iEvent.luminosityBlock() << " Bunch " 
+			       << iEvent.bunchCrossing() << std::endl;
+#endif
+  //Step1: Find if the event passes the chosen trigger
+  trigger::TriggerEvent triggerEvent;
+  edm::Handle<trigger::TriggerEvent> triggerEventHandle;
+  iEvent.getByToken(tok_trigEvt_, triggerEventHandle);
+  if (!triggerEventHandle.isValid()) {
+    edm::LogWarning("HcalIsoBunch") << "Error! Can't get the product "
+				    << triggerEvent_.label() ;
+  } else {
+    triggerEvent = *(triggerEventHandle.product());
+    /////////////////////////////TriggerResults
+    edm::Handle<edm::TriggerResults> triggerResults;
+    iEvent.getByToken(tok_trigRes_, triggerResults);
+    if (triggerResults.isValid()) {
+      std::vector<std::string> modules;
```

Comment:
it seems this variable is never used (and similarly below in the code where it is repeated)
@bsunanda